### PR TITLE
Add type and shape annotation for gm.print_readable()

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86562

For 
```
def f(a, b):
    dim0 = a.shape[0] + b.shape[0]
    dim1 = a.shape[1] + b.shape[1]
    d = a.new_empty(dim0, dim1)
    return d

fx_g = make_fx(f, tracing_mode="symbolic")(torch.randn(5, 3), torch.randn(4, 3))
fx_g.print_readable()
```

Tracing with 'real' and 'fake' mode yields 
```
class f(torch.nn.Module):
    def forward(self, a_1: Tensor<f32>[5, 3], b_1: Tensor<f32>[4, 3]):
        
        # No stacktrace found for following nodes 
        new_empty: Tensor<f32>[9, 6] = torch.ops.aten.new_empty.default(a_1, [9, 6], dtype = torch.float32, layout = torch.strided, device = device(type='cpu'), pin_memory = False);  a_1 = None
        return new_empty
```

Tracing with 'symbolic' mode yields 
```
    def forward(self, a_1: Tensor<f32>[t0.size(0), t0.size(1)], b_1: Tensor<f32>[t1.size(0), t0.size(1)]):
        
        # No stacktrace found for following nodes 
        sym_size: Symint(t0.size(0)) = torch.ops.aten.sym_size(a_1, 0)
        sym_size_1: Symint(t1.size(0)) = torch.ops.aten.sym_size(b_1, 0)
        add: Symint(t0.size(0) + t1.size(0)) = sym_size + sym_size_1;  sym_size = sym_size_1 = None
        sym_size_2: Symint(t0.size(1)) = torch.ops.aten.sym_size(a_1, 1)
        sym_size_3: Symint(t0.size(1)) = torch.ops.aten.sym_size(b_1, 1);  b_1 = None
        add_1: Symint(2*t0.size(1)) = sym_size_2 + sym_size_3;  sym_size_2 = sym_size_3 = None
        new_empty: Tensor<f32>[t0.size(0) + t1.size(0), 2*t0.size(1)] = torch.ops.aten.new_empty.default(a_1, [add, add_1], dtype = torch.float32, layout = torch.strided, device = device(type='cpu'), pin_memory = False);  a_1 = add = add_1 = None
        return new_empty
```

### CODE REVIEW

**Review by Chillee — COMMENTED** [review]
This looks super great :)

Would be nice to add a test.

I do have one question - is it an issue that we're now printing out code that isn't runnable? Do we need to add an additional object somewhere to allow us to actually run this code? Or do folks think it's fine for `print_readable` to not necessarily result in runnable code.

Also, for some bikeshedding. I think I'd prefer it if we dropped the Tensor prefix by default since the vast majority of operators are tensors.

**Comment by SherlockNoMad** [comment]
> it an issue that we're now printing out code that isn't runnable? Do we need to add an additional object somewhere to allow us to actually run this code? Or do folks think it's fine for `print_readable` to not necessarily result in runnable code.

**Comment by eellison** [comment]
Looks great 😍😍😍 

> new_empty: Tensor<f32>[t0.size(0) + t1.size(0), 2*t0.size(1)]

For this - could we just print `add_1` - since this variable is already defined elsewhere in the graph, and might be arbitrarily complex?  

+ 1 to having it be valid python - maybe something like, 
`new_empty: TT[f32, (9, 6)] = `

What do folks think about including strides for noncontiguous tensors ? Or at least hiding that beyond global verbosity config ?

**Comment by ezyang** [comment]
I would prefer a more compact syntax: f32[2,3,4] (add tuple parens or quotes if necessary as variadic generics not supported)

**Comment by Chillee** [comment]
@ezyang Should we care if the printout is actually runnable?

**Comment by ezyang** [comment]
I'm willing to give this up, but I'd like it to at least in principle be runnable on a modern enough Python. At the very least as a way to add constraints to the bikeshedding

File: torch/fx/graph.py [hunk 1]
```diff
@@ -455,8 +470,27 @@ def append_stacktrace_summary(node : Node):
...
+                # override annotation with more detailed information
+                from torch._subclasses.fake_tensor import FakeTensor
+                from torch.fx.experimental.symbolic_shapes import PySymInt
+                from torch.fx.passes.shape_prop import TensorMetadata
+
+                meta_val = node.meta.get('val', node.meta.get('tensor_meta', None))
+
+                if isinstance(meta_val, FakeTensor):
+                    maybe_type_annotation = f': Tensor<{dtype_abbrs[meta_val.dtype]}>{stringify_shape(meta_val.shape)}'
+                elif isinstance(meta_val, PySymInt):
```
Comment (wconstab) [inline torch/fx/graph.py:489]: update to include PySymFloat too

see https://github.com/pytorch/pytorch/pull/86425/files#r991657516 for some infra to leverage

**Review by Chillee — APPROVED** [review]
LGTM :)

**Pull request closed without merging**